tests/run-macros/quoted-liftable-derivation-macro-2/Derivation_1.scala

```diff
@@ -0,0 +1,77 @@
+import scala.compiletime.{erasedValue, summonFrom}
+import scala.deriving._
+import scala.quoted._
+
+trait Lft[T]:
+  def toExpr(x: T)(using Quotes): Expr[T]
+
+object Lft {
+  given Lft[Int] with
+    def toExpr(x: Int)(using Quotes) = Expr(x)
+
+  inline given derived[T](using inline m: Mirror.Of[T]): Lft[T] = ${ derivedExpr('m) }
+
+  private def derivedExpr[T](mirrorExpr: Expr[Mirror.Of[T]])(using Quotes, Type[T]): Expr[Lft[T]] = {
+    val tpe = summonExprOrError[Type[T]]
+    mirrorExpr match {
+      case '{ $mirrorExpr : Mirror.Sum { type MirroredElemTypes = mirroredElemTypes } } =>
+        val liftables = elemTypesLfts[mirroredElemTypes]
+        val liftablesFn = '{ (x: Int) => ${ switchExpr('x, liftables) } }
+        '{ new LiftableSum[T, mirroredElemTypes]($mirrorExpr, $liftablesFn)(using $tpe) }
+      case '{ $mirrorExpr : Mirror.Product { type MirroredElemTypes = mirroredElemTypes } } =>
+        val liftableExprs = Expr.ofSeq(elemTypesLfts[mirroredElemTypes])
+        '{ new LiftableProduct[T, mirroredElemTypes]($mirrorExpr, $liftableExprs)(using $tpe) }
+    }
```

Comment:
Beautiful usage of quoted pattern matches 👍

Reply:
The last thing is to be able to write `enum Opt[+T] derives ToExpr`.